server/src/services/completion/onCompletion.ts

```diff
@@ -127,7 +128,7 @@ export function doComplete(
 
     for (const orphanNode of documentAnalyzer.orphanNodes) {
       if (isNodePosition(orphanNode, newPosition)) {
```

Comment:
Nice catch here, I find the orphan nodes difficult to follow. It was only because your change forced me to read the comment that I got it:

```
  // Check if the definitionNode exists and if not, we will check if maybe Node exists in orphan Nodes.
  // This is important for "this", "super" and **global variables** because they exist only in orphanNodes.
 ```

Though I still don't have a good mental model of why global variables only exist in orphan nodes.